concordium-consensus/src/Concordium/Skov/Update.hs

```diff
@@ -517,47 +507,54 @@ doReceiveTransaction tr slot = unlessShutDown $ do
                -- but this should be revised to add a "accountExists" function in the future
                senderExists <- flip getAccount (transactionSender tx) =<< queryBlockState =<< lastFinalizedBlock
                if isNothing senderExists then return ResultNonexistingSenderAccount
-               else snd <$> doReceiveTransactionInternal tr slot
-        _ -> snd <$> doReceiveTransactionInternal tr slot
+               else snd <$> doReceiveTransactionInternal tr (utcTimeToTimestamp now) slot
+        _ -> snd <$> doReceiveTransactionInternal tr (utcTimeToTimestamp now) slot
```

Comment:
So if a credential has an invalid identity provider we are still storing it, but not retransmitting it. Is that the intended behaviour? Ideally we'd just not store it at all since it's unlikely we'd use it.

Reply:
Yea, I think the current logic is faulty. I'll clean it up.